src/modules/register/Register.jsx

```diff
@@ -252,210 +240,124 @@ class RegisterForm extends React.Component {
-	}
-	focusPasswordInput= () =>{
-		console.log(this.inputs.password.current.focus());
-	}
+	getNextScreen = () => this.nextScreen[this.state.active];
+
+	getPrevScreen = () => this.prevScreen[this.state.active];
 
-	//Progress Through Input Active States
 	showNext = () => {
-		if (this.state.nameActive) {
-			this.setState({
-				nameActive: false,
-				emailActive: true
-			});
-			this.focusEmailInput();
-		} else if (this.state.emailActive) {
-			this.setState({
-				emailActive: false,
-				passwordActive: true
-			});
-			this.focusPasswordInput();
-		}
+		this.setState({active: this.getNextScreen()});
 	};
 
 	goBack = () => {
-		if (this.state.emailActive) {
-			this.setState({
-				nameActive: true,
-				emailActive: false
-			});
-			this.focusNameInput();
-		} else if (this.state.passwordActive) {
-			this.setState({
-				emailActive: true,
-				passwordActive: false
-			});
-			this.focusEmailInput();
-		}
+		this.setState({active: this.getPrevScreen()});
 	};
 
 	// Password Show / Hide
-	togglePassword= e => {
+	togglePassword = e => {
 		this.setState({
 			showPassword: !this.state.showPassword
 		});
 	};
 
-	onChange = field => ({target: { value }})  => {
-  this.setState({
-    [field]: value,
-  })
-};
+	onChange = field => ({target: {value}}) => {
+		this.setState({
+			[field]: value
+		});
+	};
 
 	render() {
-		var status = {
-		  name: '',
-		  email: '',
-		  password: '',
-			submit: '',
-			next: '',
-			back: ''
-		}
-		var page = {
-				number: '',
-		}
-		if (this.state.nameActive) {
-			status.name = ' active';
-			page.number = '1';
-		} else {
-			status.name = '';
-		}
-		if (this.state.emailActive) {
-			status.email = ' active';
-			page.number = '2';
-		} else {
-			status.email = '';
-		}
-		if (this.state.passwordActive) {
-			status.password = ' active';
-			page.number = '3';
-			status.submit = ' show-submit';
-		} else {
-			status.password = '';
-			status.submit = '';
-		}
-		if (this.state.nameActive || this.state.emailActive) {
-			status.next = ' show-next';
-		} else {
-			status.next = '';
-		}
-		if (this.state.emailActive || this.state.passwordActive) {
-			status.back = ' show-back';
-		} else {
-			status.back = '';
-		}
+		const {active, showPassword} = this.state;
+
+		const inputProps = {
+			[this.SCREENS.NAME]: {
+				type: 'text',
+				name: "What's your name?",
+				autoComplete: 'given-name',
+				value: this.state.name
+			},
+			[this.SCREENS.EMAIL]: {
+				type: 'email',
+				name: "What's your email?",
+				autoComplete: 'email',
+				value: this.state.email
+			},
+			[this.SCREENS.PASSWORD]: {
+				type: showPassword ? 'text' : 'password',
+				name: 'Create your password',
+				autoComplete: 'current-password',
+				value: this.state.password,
+				children: (
+					<CheckBoxWrapper className="password-toggle">
+						<input className="eye-toggle" type="button" name="showPassword" id="showPassword" onClick={this.togglePassword} />
+					</CheckBoxWrapper>
+				)
+			}
+		};
```

Comment:
So here we're just grouping in all of the props that differ between the screens so that `inputProps[this.SCREENS.SCREEN_NAME]` gives us the props for that screen, and we can just pass them onto the object with 
```jsx
<Input
  {...inputProps[active]}
/>
```